config/v3_0_experimental/types/filesystem.go

```diff
@@ -15,131 +15,60 @@
-				Message: errors.ErrWarningCreateDeprecated.Error(),
-				Kind:    report.EntryWarning,
-			})
-		}
-	}
-	return r
+func (f Filesystem) ValidatePath() (r report.Report) {
+	r.AddError(validatePath(f.Path))
+	return
 }
 
-func (f Filesystem) ValidatePath() report.Report {
-	r := report.Report{}
-	if f.Path != nil && validatePath(*f.Path) != nil {
-		r.Add(report.Entry{
-			Message: fmt.Sprintf("filesystem %q: path not absolute", f.Name),
-			Kind:    report.EntryError,
-		})
-	}
-	return r
+func (f Filesystem) ValidateDevice() (r report.Report) {
+	r.AddError(validatePath(f.Device))
+	return
 }
 
-func (m Mount) Validate() report.Report {
-	r := report.Report{}
-	switch m.Format {
+func (f Filesystem) ValidateFormat() (r report.Report) {
+	switch f.Format {
 	case "ext4", "btrfs", "xfs", "swap", "vfat":
 	default:
-		r.Add(report.Entry{
-			Message: errors.ErrFilesystemInvalidFormat.Error(),
-			Kind:    report.EntryError,
-		})
-	}
-	return r
-}
-
-func (m Mount) ValidateDevice() report.Report {
-	r := report.Report{}
-	if err := validatePath(m.Device); err != nil {
-		r.Add(report.Entry{
-			Message: err.Error(),
-			Kind:    report.EntryError,
-		})
+		r.AddError(errors.ErrFilesystemInvalidFormat)
 	}
-	return r
+	return
 }
 
-func (m Mount) ValidateLabel() report.Report {
-	r := report.Report{}
-	if m.Label == nil {
-		return r
+func (f Filesystem) ValidateLabel() (r report.Report) {
+	if f.Label == nil || *f.Label == "" {
+		return
 	}
-	switch m.Format {
+	switch f.Format {
 	case "ext4":
-		if len(*m.Label) > 16 {
+		if len(*f.Label) > 16 {
 			// source: man mkfs.ext4
-			r.Add(report.Entry{
-				Message: errors.ErrExt4LabelTooLong.Error(),
-				Kind:    report.EntryError,
-			})
+			r.AddError(errors.ErrExt4LabelTooLong)
 		}
 	case "btrfs":
-		if len(*m.Label) > 256 {
+		if len(*f.Label) > 256 {
 			// source: man mkfs.btrfs
-			r.Add(report.Entry{
-				Message: errors.ErrBtrfsLabelTooLong.Error(),
-				Kind:    report.EntryError,
-			})
+			r.AddError(errors.ErrBtrfsLabelTooLong)
 		}
 	case "xfs":
-		if len(*m.Label) > 12 {
+		if len(*f.Label) > 12 {
 			// source: man mkfs.xfs
-			r.Add(report.Entry{
-				Message: errors.ErrXfsLabelTooLong.Error(),
-				Kind:    report.EntryError,
-			})
+			r.AddError(errors.ErrXfsLabelTooLong)
 		}
 	case "swap":
 		// mkswap's man page does not state a limit on label size, but through
 		// experimentation it appears that mkswap will truncate long labels to
 		// 15 characters, so let's enforce that.
-		if len(*m.Label) > 15 {
-			r.Add(report.Entry{
-				Message: errors.ErrSwapLabelTooLong.Error(),
-				Kind:    report.EntryError,
-			})
+		if len(*f.Label) > 15 {
+			r.AddError(errors.ErrSwapLabelTooLong)
 		}
 	case "vfat":
-		if len(*m.Label) > 11 {
+		if len(*f.Label) > 11 {
 			// source: man mkfs.fat
-			r.Add(report.Entry{
-				Message: errors.ErrVfatLabelTooLong.Error(),
-				Kind:    report.EntryError,
-			})
+			r.AddError(errors.ErrVfatLabelTooLong)
 		}
 	}
 	return r
```

Comment:
Can be swapped to just `return` which `r` being defined as the returned value in the function declaration now.